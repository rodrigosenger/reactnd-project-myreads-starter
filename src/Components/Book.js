import React from 'react';
import {PropTypes} from 'prop-types';
import BookShelfChanger from './BookShelfChanger';

class Book extends React.Component {
  static propTypes = {
    book: PropTypes.object.isRequired,
    changeShelf: PropTypes.func.isRequired
  }

  render() {
    const {book, changeShelf} = this.props;
    return (
      <div className="book">
        <div className="book-top">
          <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url(${book.imageLinks.thumbnail})` }}></div>
          <BookShelfChanger
            book={book}
            changeShelf={changeShelf}
          />
        </div>
        <div className="book-title">{book.title}</div>
        <div className="book-authors">{book.authors}</div>
      </div>
    )
  }
}

export default Book;