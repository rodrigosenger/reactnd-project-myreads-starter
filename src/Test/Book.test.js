import React from 'react';
import Book from '../Components/Book';
import renderer from 'react-test-renderer';

const changeBookShelf = (newBook, newShelf) => {
  update(newBook, newShelf).then(response => {
      newBook.shelf = newShelf;

      let updatedBooks = this.state.books.filter(book => book.id !== newBook.id);
      updatedBooks.push(newBook);
      this.setState({ books: updatedBooks })
  })
}

test('Rendering a book', () => {
  const book = {
		title: "Universal Artificial Intelligence",
    subtitle: "Sequential Decisions Based on Algorithmic Probability",
		authors: ["Marcus Hutter"],
		publisher: "Springer Science & Business Media",
		publishedDate: "2006-01-17",
		description: "Personal motivation. The dream of creating artificial devices that reach or outperform human inteUigence is an old one. It is also one of the dreams of my youth, which have never left me. What makes this challenge so interesting? A solution would have enormous implications on our society, and there are reasons to believe that the AI problem can be solved in my expected lifetime. So, it's worth sticking to it for a lifetime, even if it takes 30 years or so to reap the benefits. The AI problem. The science of artificial intelligence (AI) may be defined as the construction of intelligent systems and their analysis. A natural definition of a system is anything that has an input and an output stream. Intelligence is more complicated. It can have many faces like creativity, solving prob lems, pattern recognition, classification, learning, induction, deduction, build ing analogies, optimization, surviving in an environment, language processing, and knowledge. A formal definition incorporating every aspect of intelligence, however, seems difficult. Most, if not all known facets of intelligence can be formulated as goal driven or, more precisely, as maximizing some utility func tion. It is, therefore, sufficient to study goal-driven AI; e. g. the (biological) goal of animals and humans is to survive and spread. The goal of AI systems should be to be useful to humans.",
		industryIdentifiers: [{
			type: "ISBN_13",
			identifier: "9783540268772"
		}, {
			type: "ISBN_10",
			identifier: "3540268774"
		}],
		readingModes: {
			text: false,
			image: true
		},
		pageCount: 278,
		printType: "BOOK",
		categories: ["Computers"],
		maturityRating: "NOT_MATURE",
		allowAnonLogging: true,
		contentVersion: "1.4.1.0.preview.1",
		panelizationSummary: {
			containsEpubBubbles: false,
			containsImageBubbles: false
		},
		imageLinks: {
			smallThumbnail: "http://books.google.com/books/content?id=fhoonlqqGP8C&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
			thumbnail: "http://books.google.com/books/content?id=fhoonlqqGP8C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
		},
		language: "en",
		previewLink: "http://books.google.com/books?id=fhoonlqqGP8C&printsec=frontcover&dq=artificial+intelligence&hl=&cd=5&source=gbs_api",
		infoLink: "https://play.google.com/store/books/details?id=fhoonlqqGP8C&source=gbs_api",
		canonicalVolumeLink: "https://market.android.com/details?id=book-fhoonlqqGP8C",
		id: "fhoonlqqGP8C",
		shelf: "currentlyReading"
  }
  
  const component = renderer.create(
    <Book book={book} changeShelf={changeBookShelf}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})