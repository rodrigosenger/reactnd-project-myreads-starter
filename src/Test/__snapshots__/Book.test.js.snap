// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering a book 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(http://books.google.com/books/content?id=fhoonlqqGP8C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api)",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        onChange={[Function]}
        value="currentlyReading"
      >
        <option
          disabled={true}
          id="move_fhoonlqqGP8C"
          value="move"
        >
          Move to...
        </option>
        <option
          id="currentlyReading_fhoonlqqGP8C"
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          id="wantToRead_fhoonlqqGP8C"
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          id="read_fhoonlqqGP8C"
          value="read"
        >
          Read
        </option>
        <option
          id="none_fhoonlqqGP8C"
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    Universal Artificial Intelligence
  </div>
  <div
    className="book-authors"
  >
    Marcus Hutter
  </div>
</div>
`;
